File: examples/python/evm/list_token_balances.py
# Usage: uv run python evm/list_token_balances.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        address = "0x5b76f5B8fc9D700624F78208132f91AD4e61a1f0"
        network = "base-sepolia"
        token_balances = await cdp.evm.list_token_balances(address, network)

        for balance in token_balances.balances:
            print(
                f"Contract Address: {balance.token.contract_address} Balance: {balance.amount.amount}"
            )


asyncio.run(main())

File: examples/typescript/evm/listTokenBalances.ts
// Usage: pnpm tsx evm/listTokenBalances.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();

// Get the first page of token balances for barmstrong.eth.
const address = "0x5b76f5B8fc9D700624F78208132f91AD4e61a1f0";

const result = await cdp.evm.listTokenBalances({
  address,
  network: "base-sepolia",
});

for (const balance of result.balances) {
  console.log("Token contract address:", balance.token.contractAddress);
  console.log("Token balance:", balance.amount.amount);
}
